bundler/lib/bundler/cli/gem.rb

```diff
@@ -68,7 +68,7 @@ def run
         :bundler_version  => bundler_dependency_version,
         :git              => use_git,
         :github_username  => github_username.empty? ? "[USERNAME]" : github_username,
-        :required_ruby_version => Gem.ruby_version < Gem::Version.new("2.4.a") ? "2.3.0" : "2.4.0",
+        :required_ruby_version => Gem.ruby_version < Gem::Version.new("2.6.a") ? "2.3.0" : "2.6.0",
```

Comment:
This change means for Ruby 2.4 and 2.5 it will fallback to 2.3 and that is breaking some specs. To keep the old logic (for pre 2.4), we can keep the `2.4.a` check (as before), but instead of using `2.6.0`, we can use your current Ruby version.

Reply:
I see, but what if a user uses `2.5.0` for example?

Reply:
I think as long as we also change the `:linter_version` accordingly using the same criteria, namely, if using an old ruby, use an old linter version too, this change should be fine? For example, here: https://github.com/rubygems/rubygems/blob/3387dbf66d3721f1a25ea60d6bab8fddaf257f74/bundler/lib/bundler/cli/gem.rb#L169

Reply:
We could choose the latest available linter version for each Ruby version, like:
```ruby
case Gem.ruby_version
when -> v { v < Gem::Version.new("2.4.a") } then "0.81.0"
when -> v { v < Gem::Version.new("2.5.a") } then "1.12.1"
else
  "1.21.0"
end
```
If you agree we should do this, I'll add this snippet to this Pull Request. What do you think? @deivid-rodriguez

Reply:
Sure, just make sure you use `1.12` and `1.21`. We use the`~>` operator in the template so that the result is flexible enough to not require frequent updates.

Also, style wise I would stick to `if/elsif/else` rather than `case` with procs.

Reply:
@deivid-rodriguez I got it.
I'll apply the same strategy for `required_ruby_version` so that we can use the current Ruby version as a required Ruby version.

Reply:
`required_ruby_version` uses the `>=` operator by default, so what we have currently seems fine?